telegramBot.js

```diff
@@ -0,0 +1,200 @@
+
+class TelegramBot {
+    constructor( teleToken ){
+
+        //URLs for http requests
+        this.baseURL = `https://api.telegram.org/bot${teleToken}/`
+        this.updateURL = `${this.baseURL}getUpdates`
+
+        this.bot = new Telegraf(teleToken);
+
+        this.initialzied = false;
+        
+        // allows bot to obtain chat id to send messages and files
+        this.bot.command('init', (ctx) =>{
+
+            try{
+                
+                this.chatID = ctx.chat.id
+                this.chatTitle = ctx.chat.title
+                
+                ctx.reply(`Bot Initialized for Chat: ${this.chatID}` )
+                this.initialzied = true;
+
+            } catch (err) {
+                console.log('Bot Initialization Error. Unable to obtain Chat ID')
+                ctx.reply('Bot Initialization Error. Unable to obtain Chat ID')
+            }
+            
+
+        })
+
+        this.bot.command('status', (ctx) =>{
+            if(this.initialzied){
+                ctx.reply(`Bot Initialized for Chat: ${this.chatID}`)
+            } else { 
+                ctx.reply('Bot uninitialized for this Chat. Run the /init command to initialize.')
+            }
+        })
+
+        this.bot.command('help', (ctx) =>{
+            const helpText = `You can control me by sending these commands:\n
+                                \t/init - initialize bot\n
+                                \t/status - check status of bot\n
+                                \t/gm <platform> <message> - sends text message to chosen platform\n`
+
+            ctx.reply(helpText)
+        })
+
+        this.bot.command('quit', (ctx) => {
+            ctx.leaveChat()
+        })
+        
+        this.bot.command('gm', async (ctx) => {
+            if(this.initialzied) {
+
+                if(this.onCommandReceived){
+                    const input = ctx.message.text.slice(1).trim().split(' ');
+                    const command = input.shift();
+                    const commandArgs = input.join(' ');
+                    const result = await this.onCommandReceived(command, commandArgs, 'telegram');
+                    console.log(result);
+                    return result;
+                }
+                
+            } else {
+                ctx.reply('Bot uninitialized. Run /init command to initialize.')
+            }
+        })
+        
+        this.bot.command('fs', async (ctx) => {
+            if(this.initialzied)  {
+                if(this.onCommandReceived){
+                    const input = ctx.message.text.slice(1).trim().split(' ');
+                    const command = input.shift();
+                    const commandArgs = input.join(' ');
+                    const result = await this.onCommandReceived(command, commandArgs, 'telegram');
+                    console.log(result);
+                    return result;
+                }
+            
+                const fileURL = "http://i.imgur.com/5rOhtdL.png"
```

Comment:
why is the fileUrl pointing to a specific url. is this code used for testing ? for test code, I normally put it under `/test/telegramBot.test.js` and write unit test, and run `mocha /test/telegramBot.test.js`